ros/angel_system_nodes/angel_system_nodes/uho_activity_detector.py

```diff
@@ -663,7 +704,16 @@ def _process_window(self, window: InputWindow) -> ActivityDetection:
         #       [32*K x 2048]
         #       [32*K x 4]
         with SimpleTimer("Activity classification prediction", self.get_logger().info):
-            pred_conf, pred_labels = predict(self._detector, frame_set, aux_data)
+
+            if self._gt_file == "":
+                # No ground truth provided, use the detector
+                pred_conf, pred_labels = predict(self._detector, frame_set, aux_data)
+            else:
+                pred_conf, pred_labels = gt_predict(
+                    self._gt_file,
+                    time_to_int(window.frames[0][0]) * 1e-9,
+                    time_to_int(window.frames[-1][0]) * 1e-9,
+                )
```

Comment:
This isn't what I had in mind for swapping the "prediction" function. I meant that we could swap out the "window processing" function in the run-time loop. Before the loop we would assign to, say, `process_window_fn` either this method, `self._process_window`, or a ground-truth version, e.g. `self._process_window_gt`. The ground truth version of processing doesn't care about the pre- and post-processing steps that occur here. We also want to separate responsibilities for good maintainability practice for when we inevitably need to come back in here and modify the true-prediction side of things.

Reply:
Done